transport/http/middleware_content_length.go

```diff
@@ -34,17 +33,11 @@ func (m *ContentLengthMiddleware) HandleBuild(
 		return out, metadata, fmt.Errorf("unknown request type %T", req)
 	}
 
-	// Don't set content length if header is already set.
-	if vs := req.Header.Values("Content-Length"); len(vs) != 0 {
-		return next.HandleBuild(ctx, in)
-	}
-
 	if n, ok, err := req.StreamLength(); err != nil {
 		return out, metadata, fmt.Errorf(
 			"failed getting length of request stream, %w", err)
 	} else if ok && n > 0 {
-		// Only set content-length header when it is a positive value.
-		req.Header.Set("Content-Length", strconv.FormatInt(n, 10))
+		req.ContentLength = n
```

Comment:
Should this check if the `ContentLength` value  no `0` as well? and if it is not `0` don't set it? Kind of an edge case.

There is a potential conflict with the user providing a value for a `ContentLength` modeled input  member shape that conflicts with the derived length of the input stream. In similar cases like content-sha256, the middleware skips this step if the field already has a value.